doc/directives.rst

```diff
@@ -0,0 +1,185 @@
+.. _directives:
+
+Directives
+==========
+
+Hawkmoth provides several new directives for incorporating :ref:`documentation
+comments <syntax>` from C source files into the reStructuredText document. There
+are three main types of directives, for incorporating documentation from entire
+files, for single objects, and for composite objects optionally with members.
+
+Source Files
+------------
+
+The :rst:dir:`c:autodoc` directive simply includes all the documentation
+comments from any number of files. This is the most basic and quickest way to
+generate documentation, but offers no control over what gets included.
+
+.. rst:directive:: .. c:autodoc:: filename-pattern [...]
+
+   Incorporate documentation comments from the files specified by the space
+   separated list of filename patterns given as arguments. The patterns are
+   interpreted relative to the :data:`cautodoc_root` configuration option.
+
+   .. rst:directive:option:: transform
+      :type: text
+
+      Name of the transformation function specified in
+      :data:`cautodoc_transformations` to use for converting the comments. This
+      value overrides the default in :data:`cautodoc_transformations`.
+
+   .. rst:directive:option:: compat
+      :type: text
+
+      The ``compat`` option overrides the :data:`cautodoc_compat` configuration
+      option.
+
+      .. warning::
+
+	 The compat option has been deprecated in favour of the
+	 :data:`cautodoc_transformations` option and the :rst:dir:`c:autodoc`
+	 directive ``transform`` option, and will be removed in the future.
```

Comment:
As long as we're here, a minor nitpick: this compiles fine, but the indentation mixes spaces and tabs. Looks good with 8 space wide tabs, but that is unconventional in rst / Python.

Reply:
Right, Emacs likes to use tabs for rst mode, maybe I need to fix some config. Thanks, fixed locally.

Reply:
Also added a commit to remove the two remaining tabs from `extension.rst`.